keep/action/action.py

```diff
@@ -54,9 +54,11 @@ def _check_conditions(self):
                 # TODO - wrap it else
                 if step == "this":
                     continue
-                for condition in full_context.get("steps").get(step).get("conditions"):
+                for condition, condition_results in (
+                    full_context.get("steps").get(step).get("conditions").items()
+                ):
                     # One of the conditions has been met
-                    if condition.get("result"):
+                    if any([c["result"] for c in condition_results]):
```

Comment:
i'm not 100% sure I understand this one

Reply:
there was a bug in the previous code - it assumed that there is one condition result per condition, which is not the case for "foreach". anyway we are going to rewrite the conditions (get them out of steps)